ksql-execution/src/main/java/io/confluent/ksql/execution/plan/KTableHolder.java

```diff
@@ -16,19 +16,32 @@
 package io.confluent.ksql.execution.plan;
 
 import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.execution.materialization.MaterializationInfo;
 import java.util.Objects;
+import java.util.Optional;
 import org.apache.kafka.streams.kstream.KTable;
 
 public final class KTableHolder<K> {
   private final KTable<K, GenericRow> stream;
   private final KeySerdeFactory<K> keySerdeFactory;
+  private final Optional<MaterializationInfo.Builder> materializationBuilder;
 
   public KTableHolder(
       final KTable<K, GenericRow> stream,
-      final KeySerdeFactory<K> keySerdeFactory
+      final KeySerdeFactory<K> keySerdeFactory,
+      final Optional<MaterializationInfo.Builder> materializationBuilder
   ) {
     this.stream = Objects.requireNonNull(stream, "stream");
     this.keySerdeFactory = Objects.requireNonNull(keySerdeFactory, "keySerdeFactory");
+    this.materializationBuilder =
+        Objects.requireNonNull(materializationBuilder, "materializationProvider");
+  }
+
+  public KTableHolder(
```

Comment:
Do we need the overloaded constructor?  Such constructors can make it hard for people to know _which_ to call.  Consider just inlining this constructor.